plugins/module_utils/foreman_helper.py

```diff
@@ -203,7 +205,7 @@ def __init__(self, **kwargs):
 
         self.validate_parameters()
 
-    def run(self, **kwargs):
+    def update_parameters(self):
```

Comment:
Just curious -- is there a direct technical benefit to separating the logic into separate `update_parameters` and `run` methods here, or is the benefit "limited to" better conceptual clarity and readability?

Reply:
If you look at the very first commit of the series, it added code to this function and I needed that code to run at the very beginning of HostMixin.run, not at the end (as it would happen when HM.run calls ParameterMixin.run). With the latest iteration it's probably not required anymore and can be reverted back.